### PR TITLE
Update Elasticity config for aws session token

### PR DESCRIPTION
**Pull request by 13scoobie**
Updated the Elasticity config to support pulling in InstanceProfileCredentials if config is set to iam, else, pull from config for user + access key

### CODE REVIEW

File: 3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/emr_job.rb [hunk 1]
```diff
@@ -102,8 +102,18 @@ def initialize(debug, enrich, shred, elasticsearch, s3distcp, config, enrichment
 
         # Configure Elasticity with your AWS credentials
         Elasticity.configure do |c|
-          c.access_key = config[:aws][:access_key_id]
-          c.secret_key = config[:aws][:secret_access_key]
+          aws_session_token = nil
+ +        if config[:aws][:access_key_id] == 'iam' and config[:aws][:secret_access_key] == 'iam'
+            #federated Identity Management
+ +          credentials_from_role = Aws::InstanceProfileCredentials.new.credentials
```
Comment (seanhall) [inline 3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/emr_job.rb:108]: Don't forget to add the aws-sdk gem to your Gemfile and Gemfile.lock, same as https://github.com/snowplow/snowplow/commit/c77b8f71671ccf600088899673e0efcd8321ea8e.

File: 3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/emr_job.rb [hunk 2]
```diff
@@ -102,8 +102,18 @@ def initialize(debug, enrich, shred, elasticsearch, s3distcp, config, enrichment
 
         # Configure Elasticity with your AWS credentials
         Elasticity.configure do |c|
-          c.access_key = config[:aws][:access_key_id]
-          c.secret_key = config[:aws][:secret_access_key]
+          aws_session_token = nil
```
Comment (seanhall) [inline 3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/emr_job.rb]: This doesn't hurt, but I don't think it's doing anything or is needed.

**Comment by seanhall** [comment]
Just 1-2 small changes, and I think this is good to go.  Nice!!

File: 3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/emr_job.rb (hunk 1 above)
Comment (13scoobie) [inline 3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/emr_job.rb:108]: Good call!

**Comment by seanhall** [comment]
~~Almost forgot contracts.rb and s3_tasks.rb.  https://jira.nordstrom.net/browse/ATO-4399  But this is a great start and template!~~

**Pull request merged**